scripts/fetch-ui-assets.sh

```diff
@@ -12,6 +12,21 @@
 # respective releases in "influxdata/ui" (OSS-2.0, OSS-2.1, etc). Those releases
 # are updated only when a bug fix needs included for the UI of that OSS release.
 
-curl -L https://github.com/influxdata/ui/releases/download/OSS-Master/build.tar.gz --output ui/build.tar.gz
-tar -xzf ui/build.tar.gz -C ui
-rm ui/build.tar.gz
+set -e
+
+# Download the SHA256 checksum attached to the release. To verify the integrity
+# of the download, this checksum will be used to check the download tar file
+# containing the built UI assets.
+curl -Ls https://github.com/influxdata/ui/releases/download/OSS-Master/sha256.txt --output sha256.txt
```

Comment:
I'd expect the sha to be hard-coded here instead of pulled from a `curl` (like in [this script](https://github.com/influxdata/influxdb/blob/master/scripts/ci/install-rust.sh)). Otherwise there's still a security risk: if attackers can compromise the `build.tar.gz` download, they can also compromise the `sha256.txt` download.

Reply:
Indeed...this is really an integrity check (did the file get corrupted in transit?) rather than a strict security check. That's how I interpreted the intent of the referenced issue (https://github.com/influxdata/idpe/issues/10889) - maybe @pierwill has some other thoughts.

To hard-code the sha we'd have to version the `OSS-Master` tag somehow which I can see the appeal of, but it would make keeping the `master` branch of the two repos in sync a little more involved. I had hoped to some day automatically update the `OSS-Master` release on pushes to the UI repo `master` branch, although I'm not sure how likely it is that will ever happen.

It would be very easy to hard-code the sha on release branches, since those are pointed at specific tagged releases in `ui`, (ex: [for 2.0.7](https://github.com/influxdata/influxdb/blob/c72285b20a2d96b997ed7c2691154e6998d01f70/ui/fetch_ui_assets.sh#L15)), so we should definitely do that on our next release branch.

I'd have to think more about how it would work to version the UI assets from the `ui` repo master branch if we want to hard code the SHA on the `influxdb` `master` branch as well though.

Reply:
Great discussion here! Thanks for taking this on, @wbaker85.

Now: do we hardcode or fetch the SHA? As always, there are tradeoffs within tradeoffs. I agree @wbaker85, this PR does a basic integrity check, which is great. @danxmoran 's point that hardcoding the SHA makes certain kinds of malicious alterations more difficult.

Sounds like there's an engineering cost to hardcoding the SHA in a way that will work smoothly. If we have the bandwidth, I think it'd be a nice boost to the security of the build pipeline. But then, there are maintenance costs of updating the SHA...

Suggestion: we merge this PR, and look into ways to secure this further.